test/jobs/calendar_importer/event_resolver_strategy_test.rb

```diff
@@ -0,0 +1,275 @@
+
+    @notices = []
+    @from_date = Date.new(1990, 1, 1)
+  end
+
+  def test_event_strategy_with_data_location_with_place_uses_partner_place
+    # theory of test:
+    #   given
+    #     data location is present
+    #     calendar strategy is 'event'
+    #
+    #   then
+    #     use location from data
+    #       (not from calendar)
+
+    data = FakeEvent.new(
+      uid: 123,
+      summary: 'A summary',
+      description: 'A description',
+      location: @address_partner.name,
+      rrule: '',
+      last_modified: '',
+      ocurrences_between: [[@start_date, @end_date]],
+      has_location?: true,
+      postcode: ''
+    )
+
+    calendar = create(:calendar, strategy: 'event')
+
+    resolver = CalendarImporter::EventResolver.new(data, calendar, @notices, @from_date)
+
+    place, address = resolver.event_strategy(calendar.place)
+    assert_equal place, @address_partner
+  end
+
+  def test_event_overide_strategy_with_data_location_with_place_uses_partner_place
+    # (same as above?)
+
+    # theory of test:
+    #   given
+    #     data location is present
+    #     calendar strategy is 'event_override'
+    #   then
+    #     use location from data
+    #       (not from calendar)
+
+    data = FakeEvent.new(
+      uid: 123,
+      summary: 'A summary',
+      description: 'A description',
+      location: @address_partner.name,
+      rrule: '',
+      last_modified: '',
+      ocurrences_between: [[@start_date, @end_date]],
+      has_location?: true,
+      postcode: ''
+    )
+
+    calendar = create(:calendar, strategy: 'event_override') # <--- different strategy
+
+    resolver = CalendarImporter::EventResolver.new(data, calendar, @notices, @from_date)
+
+    place, address = resolver.event_strategy(calendar.place)
+    assert_equal place, @address_partner
+  end
+
+  def test_place_strategy_works
+    # theory of test:
+    #   given
+    #     data location is address
+    #     calendar strategy is 'place'
+    #
+    #   then
+    #     event address = data address
+    #     event place = calendar place
+
+    data = FakeEvent.new(
+      uid: 123,
+      summary: 'A summary',
+      description: 'A description',
+      location: @address_partner.name,
+      rrule: '',
+      last_modified: '',
+      ocurrences_between: [[@start_date, @end_date]],
+      has_location?: true,
+      postcode: ''
+    )
+
+    calendar = create(:calendar, strategy: 'place')
+
+    resolver = CalendarImporter::EventResolver.new(data, calendar, @notices, @from_date)
+
+    place, address = resolver.event_strategy(calendar.place)
+    assert_equal place, @address_partner
+  end
+
+  def test_override_strategy_works
+    # theory of test
+    #   given
+    #     data location is missing
+    #     calendar strategy is 'event_override'
+    #
+    #   then
+    #     event place = calendar place
+    #     event address = calendar place address
+
+    data = FakeEvent.new(
```

Comment:
If you're using the same `data` for each test and only at most changing one or two values: I would put it into the setup and change that value on the struct in the test. This should just make it a little bit clearer that it is either the same structure, or mostly the same :)

Reply:
fixed